### PR TITLE
Require options where they're used

### PR DESCRIPTION
**Pull request by honzajavorek**
#### :rocket: Why this change?

Streamlining how the codebase is written.

#### :memo: Related issues and Pull Requests

Factored out from https://github.com/apiaryio/dredd/pull/1289/

#### :white_check_mark: What didn't I forget?

- [ ] To write docs
- [ ] To write tests
- [x] To put [Conventional Changelog](https://dredd.org/en/latest/internals.html#sem-rel) prefixes in front of all my commits and run `npm run lint`

### CODE REVIEW

File: lib/CLI.js [hunk 1]
```diff
@@ -13,8 +13,10 @@ const logger = require('./logger');
 const { applyLoggingOptions } = require('./configuration');
 const { spawn } = require('./childProcess');
 
+const dreddOptions = require('./options.json');
```
Comment (kylef) [inline lib/CLI.js]: I don't think the extension here is nessecery, in same way how you don't specify `.js` on the other sources. We can omit for consistency.

```suggestion
const dreddOptions = require('./options');
```

File: lib/Dredd.js [hunk 2]
```diff
@@ -250,5 +249,5 @@ Is the provided path correct?
   }
 }
 
+
 module.exports = Dredd;
-module.exports.options = options;
```
Comment (kylef) [inline lib/Dredd.js]: Is this part of the public API / a breaking change?

Comment (honzajavorek) [reply lib/Dredd.js]: Hmm, you're right. While it's undocumented, this is breaking, I'll amend the message. I want it removed as I don't see any value in having it exported to users like this, the options definitions are internal to Dredd.

File: lib/CLI.js (hunk 1 above)
Comment (honzajavorek) [reply lib/CLI.js]: Oh, I thought it is necessary for `.json`. I don't know why.

**Comment by honzajavorek** [comment]
Waiting for https://github.com/apiaryio/dredd/pull/1294 to get merged first.

**Pull request merged**